Classes/Hooks/DataHandler.php

```diff
@@ -0,0 +1,156 @@
+	*
+	* Structure of the deprecatedFields array:
+	*
+	* [ 'list_type' => ['field1', 'field2'] ];
+	*
+	* @param string $status
+	* @param string $table
+	* @param string $id
+	* @param array $fieldArray
+	* @param \TYPO3\CMS\Core\DataHandling\DataHandler $pObj
+	*
+	* @return void
+	*/
+	public function processDatamap_postProcessFieldArray( $status, $table, $id, array &$fieldArray, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj )
+	{
+		// Only listen to flexform data
+		if( $table === 'tt_content' && $status === 'update' && isset( $fieldArray['pi_flexform'] ) )
+		{
+			// Get list type
+			$listType = isset( $fieldArray['list_type'] ) ? $fieldArray['list_type'] : '';
+
+			if( empty( $listType ) )
+			{
+				$record = BackendUtility::getRecord( $table, $id, 'list_type' );
+				$listType = $record['list_type'];
+			}
+
+
+			// Only listen to own plugin data
+			if( !empty( $listType ) && compare( $listType, 'aimeos_', 1) == 0 )
+			{
+				// Define deprecated fields which should be ignored on saving
+				$deprecatedFields = array(
+					/*
+					// Links to deprecated plugins
+					'aimeos_catalog-detail' => array(
+						'settings.client.html.catalog.stock.url.target',
+					),
+					'aimeos_catalog-filter' => array(
+						'settings.client.html.catalog.count.url.target',
+						'settings.client.html.catalog.suggest.url.target',
+					),
+					'aimeos_catalog-list' => array(
+						'settings.client.html.catalog.stock.url.target',
+					),
+					'aimeos_checkout-standard' => array(
+						'settings.client.html.checkout.update.url.target',
+					),
+
+					// Deprecated plugins
+					'aimeos_catalog-count' => array(
+						'settings.typo3.tsconfig',
+					),
+					'aimeos_catalog-stock' => array(
+						'settings.typo3.tsconfig',
+					),
+					'aimeos_catalog-suggest' => array(
+						'settings.client.html.catalog.detail.url.target',
+						'settings.typo3.tsconfig',
+					),
+					'aimeos_checkout-update' => array(
+						'settings.client.html.checkout.confirm.url.target',
+						'settings.typo3.tsconfig',
+					),
+					*/
+				);
+
+
+				// Get array from flex from xml
+				$flexformData = GeneralUtility::xml2array( $fieldArray['pi_flexform'] );
+
+				// Search for deprecated plugins fields
+				foreach( $deprecatedFields as $plugin => $fields )
+				{
+					if( $listType === $plugin )
+					{
+						// Search for deprecated fields and delete
+						foreach( $fields as $field )
+						{
+							if ( isset( $flexformData['data']['sDEF']['lDEF'][$field] ) )
+							{
+								unset( $flexformData['data']['sDEF']['lDEF'][$field] );
+							}
+						}
+
+						// Remove empty sheet
+						if ( isset( $flexformData['data']['sDEF']['lDEF'] ) && $flexformData['data']['sDEF']['lDEF'] === array() )
+						{
+							unset( $flexformData['data']['sDEF'] );
+						}
+					}
+				}
+
+
+				// Search for empty fields and delete
+				foreach( $flexformData['data']['sDEF']['lDEF'] as $field => $values )

```

Comment:
Here I search for empty values of the plugins or whatever. T3 stores this even if empty. I simply unset all empty fields that they don't get stored to the db. If there are no values set, the complete sheet will be ignored / unset